array.go

```diff
@@ -208,59 +208,76 @@ func (a *Array) Close() error {
 	return nil
 }
 
-// Create creates a new TileDB array given an input schema.
-func (a *Array) Create(arraySchema *ArraySchema) error {
-	curi := C.CString(a.uri)
+// Create creates a new TileDB array given a context, URI and schema.
+func CreateArray(tdbCtx *Context, uri string, arraySchema *ArraySchema) error {
+	curi := C.CString(uri)
 	defer C.free(unsafe.Pointer(curi))
-	ret := C.tiledb_array_create(a.context.tiledbContext, curi, arraySchema.tiledbArraySchema)
-	runtime.KeepAlive(a)
+	ret := C.tiledb_array_create(tdbCtx.tiledbContext, curi, arraySchema.tiledbArraySchema)
+	runtime.KeepAlive(tdbCtx)
 	runtime.KeepAlive(arraySchema)
 	if ret != C.TILEDB_OK {
-		return fmt.Errorf("error creating tiledb array: %w", a.context.LastError())
+		return fmt.Errorf("error creating tiledb array: %w", tdbCtx.LastError())
 	}
 	return nil
 }
 
-// Consolidate consolidates the fragments of an array into a single fragment.
+// Create creates a new TileDB array given an input schema.
+func (a *Array) Create(arraySchema *ArraySchema) error {
+	return CreateArray(a.context, a.uri, arraySchema)
+}
+
+// ConsolidateArray consolidates the fragments of an array into a single fragment.
 // You must first finalize all queries to the array before consolidation can
 // begin (as consolidation temporarily acquires an exclusive lock on the array).
-func (a *Array) Consolidate(config *Config) error {
+func ConsolidateArray(tdbCtx *Context, uri string, config *Config) error {
 	if config == nil {
 		return errors.New("Config must not be nil for Consolidate")
 	}
 
-	curi := C.CString(a.uri)
+	curi := C.CString(uri)
 	defer C.free(unsafe.Pointer(curi))
-	ret := C.tiledb_array_consolidate(a.context.tiledbContext, curi, config.tiledbConfig)
-	runtime.KeepAlive(a)
+	ret := C.tiledb_array_consolidate(tdbCtx.tiledbContext, curi, config.tiledbConfig)
+	runtime.KeepAlive(tdbCtx)
 	runtime.KeepAlive(config)
 	if ret != C.TILEDB_OK {
-		return fmt.Errorf("error consolidating tiledb array: %w", a.context.LastError())
+		return fmt.Errorf("error consolidating tiledb array: %w", tdbCtx.LastError())
 	}
 
 	runtime.KeepAlive(config)
 	return nil
 }
 
-// Vacuum cleans up the array, such as consolidated fragments and array metadata.
-func (a *Array) Vacuum(config *Config) error {
+// Consolidate consolidates the fragments of the array into a single fragment.
+// You must first finalize all queries to the array before consolidation can
+// begin (as consolidation temporarily acquires an exclusive lock on the array).
```

Comment:
```suggestion
// begin (as consolidation temporarily acquires an exclusive lock on the array).
// Deprecated: Use ConsolidateArray instead.
```

Reply:
> I don't see a reason to deprecate?

The reason is the same with the `Create` APIs: there is no C API to consolidate/vacuum an already opened array. TileDB-Go so far has provided a mostly 1:1 API surface with the C API, and these methods require extra book-keeping of the URI that I would like to remove.

> a client would probably just prefix with calls to Array.URI() and Array.Context() anyhow

This might be a valid scenario for consolidation, but people really shouldn't vacuum an already open array. 😅

Reply:
Ah I see, thanks for the clarification! SGTM if you'd like to deprecate them